tests/integration/ha_tests/helpers.py

```diff
@@ -704,7 +722,14 @@ def modify_pebble_restart_delay(
                 _preload_content=False,
             )
             response.run_forever(timeout=60)
-            if ensure_replan:
+            if ensure_replan and response.returncode != 0:
+                # Juju 2 fix service is spawned but pebble is reporting inactive
+                try:
+                    await send_signal_to_process(
+                        ops_test, unit_name, "/usr/bin/patroni", "SIGTERM"
+                    )
+                except (ProcessError, ProcessRunningError):
+                    pass
```

Comment:
Juju 2 tests failed frequently here, investigating locally I saw dangling patroni process with the service being inactive. Killing the process here, so that pebble is able to restart.

Reply:
It shouldn't be dangerous, replan should do the same, but we can limit to juju 2.